### PR TITLE
AppGroup support

### PR DESCRIPTION
**Pull request by ewcchan**
https://github.com/CPE-OAuth/servicestage-eclipse-plugin/issues/20

### CODE REVIEW

File: src/main/java/com/huawei/cloud/servicestage/client/AuthClient.java [hunk 1]
```diff
@@ -65,6 +65,9 @@ public static Token getAuthToken(String region, String username,
         CloseableHttpClient client = HttpClients.createDefault();
         CloseableHttpResponse response = null;
 
+        // update domain value for non IAM users
+        domain = (domain!=null || domain=="")? domain : username;
```
Comment (paulmchen) [inline src/main/java/com/huawei/cloud/servicestage/client/AuthClient.java]: domain != null  includes the case of domain=="", isn't ?  Does it mean  line 69 should be the same as the following line?

   domain = (domain!=null )? domain : username;

File: src/main/java/com/huawei/cloud/servicestage/client/ServiceStageClient.java [hunk 2]
```diff
@@ -182,7 +185,10 @@ public SimpleResponse updateService(String instanceId, String requestBody,
         request.setHeader(X_AUTH_TOKEN_HEADER_KEY, token.getToken());
 
         // proxy (if needed)
-        Util.setProxy(request);
+        if (!requestUrl.contains("localhost")) {
```
Comment (paulmchen) [inline src/main/java/com/huawei/cloud/servicestage/client/ServiceStageClient.java]: there are too many repeating of the following lines in the code, could it be added to the Util.setProxy() method instead? 

if (!requestUrl.contains("localhost")) {
   Util.setProxy(request);
}

File: src/main/java/com/huawei/cloud/servicestage/client/AuthClient.java (hunk 1 above)
Comment (ewcchan) [reply src/main/java/com/huawei/cloud/servicestage/client/AuthClient.java]: @paulmchen  it is not the same, but I will update the code to use isEmpty() instead which checks both use cases.

File: src/main/java/com/huawei/cloud/servicestage/client/ServiceStageClient.java (hunk 2 above)
Comment (ewcchan) [reply src/main/java/com/huawei/cloud/servicestage/client/ServiceStageClient.java]: @paulmchen code is updated and logic pushed to the Util class.

**Pull request merged**